trino-aws-proxy/src/test/java/io/trino/aws/proxy/server/rest/TestProxiedErrorResponses.java

```diff
@@ -95,12 +99,21 @@ public Builder filter(Builder builder)
     }
 
     @Inject
-    public TestProxiedErrorResponses(S3Client internalClient, TestingRemoteS3Facade delegatingFacade, @ForErrorResponseTest TestingHttpServer httpErrorResponseServer)
+    public TestProxiedErrorResponses(TestingRemoteS3Facade delegatingFacade, @ForErrorResponseTest TestingHttpServer httpErrorResponseServer, @ForTesting Credentials testingCredentials)
     {
-        this.internalClient = requireNonNull(internalClient, "internal client is null");
+        internalClient = clientBuilder(httpErrorResponseServer.getBaseUrl())
+                .forcePathStyle(true)
+                .credentialsProvider(() -> AwsSessionCredentials.create(testingCredentials.emulated().accessKey(), testingCredentials.emulated().secretKey(), testingCredentials.emulated().session().orElse("")))
+                .build();
         delegatingFacade.setDelegate(new PathStyleRemoteS3Facade((_, _) -> httpErrorResponseServer.getBaseUrl().getHost(), false, Optional.of(httpErrorResponseServer.getBaseUrl().getPort())));
```

Comment:
In this state, this doesn't test anything, right?

The point was to test that aws-proxy propagates errors from the remote S3 correctly.

What this is doing currently is bypassing the aws-proxy and calling `httpErrorResponseServer` with an S3 client

Reply:
Derp - I messed it up I think. Feel free to submit a fixup PR. I probably didn't understand the test.